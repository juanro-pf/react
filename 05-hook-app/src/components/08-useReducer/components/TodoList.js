import React from 'react';
import { TodoListItem } from './TodoListItem';
import PropTypes from 'prop-types';

export const TodoList = ({todos, handleToggle, handleDelete}) => {
    return (
        <ul className='list-group list-group-flush'>
            {
                todos.map( (todo, i) => 
                    <TodoListItem
                        key= {todo.id}
                        todo= {todo}
                        i= {i}
                        handleToggle= {handleToggle}
                        handleDelete= {handleDelete}
                    />
                )
            }
        </ul>
    );
};

TodoList.propTypes= {
    todos: PropTypes.array.isRequired,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired
};