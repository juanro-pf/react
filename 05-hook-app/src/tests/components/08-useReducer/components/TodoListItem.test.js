import React from 'react';
import { demoTodos } from '../../../fixtures/demoTodos';
import { shallow } from 'enzyme';
import { TodoListItem } from '../../../../components/08-useReducer/components/TodoListItem';

import '@testing-library/jest-dom';

describe('Testing TodoListItem', () => {

    const handleToggle = jest.fn();
    const handleDelete = jest.fn();

    let wrapper= shallow(
        <TodoListItem 
            todo= {demoTodos[0]}
            i= {0}
            handleToggle= { handleToggle }
            handleDelete= { handleDelete }
        />
    );
    
    test('Snapshot', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('handleDelete', () => {
        
        wrapper.find('button').simulate('click');

        expect( handleDelete ).toHaveBeenCalledWith( demoTodos[0].id );

    });
    
    test('handleToggle', () => {
        
        wrapper.find('p').simulate('click');

        expect( handleToggle ).toHaveBeenCalledWith( demoTodos[0].id );

    });

    test('correct <p></p>', () => {
        
        const p= wrapper.find('p');

        expect( p.text().trim()).toBe( `${ 1 }. ${ demoTodos[0].desc }` );

    });

    test('complete class', () => {
        
        const todo= demoTodos[0];
        todo.done= true;

        wrapper= shallow(
            <TodoListItem 
                todo= {todo}
                i= {0}
                handleToggle= { handleToggle }
                handleDelete= { handleDelete }
            />
        );

        expect(wrapper.find('p').hasClass('complete')).toBe(true);

    });
     
});
