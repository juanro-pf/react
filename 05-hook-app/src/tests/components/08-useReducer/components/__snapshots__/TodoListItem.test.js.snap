// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TodoListItem Snapshot 1`] = `
<li
  className="list-group-item"
  key="1"
>
  <p
    className="false"
    onClick={[Function]}
  >
    1
    . 
    React
  </p>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
